sale_input_barcode_gs1/models/product_barcode_line_mixin.py

```diff
@@ -0,0 +1,61 @@
+# copyright 2022 David BEAL @ Akretion
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class ProductLineMixin(models.AbstractModel):
+    _inherit = "product.barcode.line.mixin"
+
+    @api.model
+    def _decode_barcode(self, raw_barcode):
+        """
+        res =>
+        {
+            "01": "03400933816759",  # GTIN
+            "17": "2014-05-31",      # expiry date
+            "10": "B04059A",         # lot number
+            "310": 0.06385,          # weight
+            "15": "2014-05-01",      # sale date
+        }
+        """
+        barcode_dict = self.env["gs1_barcode"].decode(raw_barcode)
```

Comment:
Missing super here, then breaking inheriting.

Reply:
Why you collapsed my comment here ?

Reply:
Apologies. I did address the missing `super` though.